client/app/src/main/java/com/example/criengine/Fragments/FilterFragment.java

```diff
@@ -0,0 +1,100 @@
+package com.example.criengine.Fragments;
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.util.SparseBooleanArray;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.AbsListView;
+import android.widget.ArrayAdapter;
+import android.widget.GridView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.DialogFragment;
+
+import com.example.criengine.R;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A generic fragment for the purpose of managing a list of filters
+ */
+public class FilterFragment extends DialogFragment implements Serializable {
+    private String title;
+    private List<String> filters;
+    private List<String> activeFilters;
+    private OnFragmentInteractionListener listener;
+
+    /**
+     * Constructor 2. Initializes all required properties
+     *
+     * @param filters The List of all possible filters
+     * @param activeFilters The List containing only the enabled filters.
+     * @param listener external action listener
+     */
+    public FilterFragment(String title, List<String> filters,
+                          List<String> activeFilters, OnFragmentInteractionListener listener) {
+        this.title = title;
+        this.filters = filters;
+        this.activeFilters = activeFilters;
+        this.listener = listener;
+    }
+
+    /**
+     * Creates the fragment.
+     *
+     * @param savedInstanceState If the activity is being re-initialized after previously being
+     *                           shut down then this Bundle contains the data it most recently
+     *                           supplied.
+     * @return The dialog allowing the user to select a filer(s).
+     */
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+        View view = LayoutInflater.from(getActivity()).inflate(R.layout.fragment_filter, null);
+
+        // Populate the layout with all the required checkboxes
+        ArrayAdapter adapter = new ArrayAdapter(getContext(),
+                android.R.layout.simple_list_item_multiple_choice, filters.toArray());
+        final GridView gridView = (GridView) view.findViewById(R.id.checkbox_grid);
+        gridView.setChoiceMode(AbsListView.CHOICE_MODE_MULTIPLE);
+        gridView.setAdapter(adapter);
+
+        // initialize the checkboxes
+        for (int i = 0; i < adapter.getCount(); i++) {
+            gridView.setItemChecked(i, activeFilters.contains(filters.get(i)));
+        }
+
+        // finalize the fragment view
+        this.setCancelable(false);

```

Comment:
Pretty sure setCancelable just makes it so that you cant cancel the fragment, not "finalize the fragment view"
but its all good, no one will notice hehe

Reply:
Oh, I thought finalize meant like "finish constructing the fragment".
like setting the tittle  and disabling  the cancel button are part of finishing the construction.
you're right though, its a minor detail